webview/src/experiments/components/table/TableHeader.tsx

```diff
@@ -184,9 +191,16 @@ const TableHeaderCellContents: React.FC<{
         onDrop={onDrop}
       />
       {canResize && (
+        /* eslint-disable jsx-a11y/no-static-element-interactions */
         <div
           {...column.getResizerProps()}
-          onMouseEnter={() => setMenuSuppressed(true)}
+          onMouseDownCapture={() => {
+            dispatch(changeIsColumnResizing(true))
+          }}
+          onMouseUp={() => dispatch(changeIsColumnResizing(false))}
```

Comment:
The tests seem to be failing due to this line but I can't figure out why. I don't think the function is mutating the state and even if it was, I don't think we need to worry about state mutation because of [how `@redux/toolkit` works ](https://redux-toolkit.js.org/api/createReducer#direct-state-mutation)🤔 

```
dvc-vscode-webview:test:     Invariant failed: A state mutation was detected between dispatches, in the path 'tableData.columnWidths.id'.  This may cause incorrect behavior. (https://redux.js.org/style-guide/style-guide#do-not-mutate-state)
dvc-vscode-webview:test: 
dvc-vscode-webview:test:       198 |             dispatch(changeIsColumnResizing(true))
dvc-vscode-webview:test:       199 |           }}
dvc-vscode-webview:test:     > 200 |           onMouseUp={() => dispatch(changeIsColumnResizing(false))}
dvc-vscode-webview:test:           |                            ^
dvc-vscode-webview:test:       201 |           onMouseEnter={() => {
dvc-vscode-webview:test:       202 |             setMenuSuppressed(true)
dvc-vscode-webview:test:       203 |           }}
```

I could also get the tests to pass by just using a `useState` hook in the `Table` component for column resizing, but that method required touching practically all table components since we needed to carry down the state to the cell components.

Reply:
Good work! Tell me please, I have noticed a delay of hover style when the cell on hover. Yes?

Reply:
> Good work! Tell me please, I have noticed a delay of hover style when the cell on hover. Yes?

I'm not seeing a delay locally. Could you explain a bit more about what you're seeing?

Reply:
Not sure, maybe it is because of the video, but I feel like when you hover on the cell there is some small delay before the hover style is executed. No?

Reply:
> Not sure, maybe it is because of the video, but I feel like when you hover on the cell there is some small delay before the hover style is executed. No?

Possibly, since we have partially rely on some JS for the column styles. Also might just be the video :)

Reply:
> The tests seem to be failing due to this line but I can't figure out why. I don't think the function is mutating the state and even if it was, I don't think we need to worry about state mutation because of [how `@redux/toolkit` works ](https://redux-toolkit.js.org/api/createReducer#direct-state-mutation)🤔
> 
> ```
> dvc-vscode-webview:test:     Invariant failed: A state mutation was detected between dispatches, in the path 'tableData.columnWidths.id'.  This may cause incorrect behavior. (https://redux.js.org/style-guide/style-guide#do-not-mutate-state)
> dvc-vscode-webview:test: 
> dvc-vscode-webview:test:       198 |             dispatch(changeIsColumnResizing(true))
> dvc-vscode-webview:test:       199 |           }}
> dvc-vscode-webview:test:     > 200 |           onMouseUp={() => dispatch(changeIsColumnResizing(false))}
> dvc-vscode-webview:test:           |                            ^
> dvc-vscode-webview:test:       201 |           onMouseEnter={() => {
> dvc-vscode-webview:test:       202 |             setMenuSuppressed(true)
> dvc-vscode-webview:test:       203 |           }}
> ```
> 
> I could also get the tests to pass by just using a `useState` hook in the `Table` component for column resizing, but that method required touching practically all table components since we needed to carry down the state to the cell components.

Let me try to run the tests locally to see what's happening

Reply:
The problem is with line 351 and 388 of the `Table.test.tsx`.  `columnWidths` ids must be passed to Redux now and it is being mutated by `columnWidths.id = 353`

Reply:
Discussed the issue with Stephanie and decided to delete the second half of the test for now since it's mutation the state and the first half should be enough.